File: best_time_to_buy_and_sell_stock.cpp
class Solution {
public:
	int maxProfit(vector<int>& prices) {
		// Your code goes here
        int minSofar = prices[0];
        int maxProfitTillHere = 0;
        for (int price : prices) {
            if (price < minSofar) {
                minSofar = price;
            } else {
                maxProfitTillHere = std::max(maxProfitTillHere, price - minSofar);
            }
        }
        return maxProfitTillHere;
	}
};

File: best_time_to_buy_and_sell_stock.js
class Solution {
	/** 
	 * @param {number[]} prices
	 * @return {number}
	 */
	maxProfit(prices) {
		// Your code goes here
        let minPrice = prices[0];
        let maxProfit = 0;
        prices.forEach((price)=>{
            if(price<minPrice){
                minPrice=price;
            }else{
                maxProfit = Math.max(price-minPrice,maxProfit);
            }
        })
        return maxProfit;
	}
};
